File: opensef/test/hello-window.cpp
/**
 * hello-window.cpp - Visual openSEF test
 *
 * Creates an actual window with Space Orange background!
 */

#include <iostream>
#include <memory>
#include <opensef/OSFWindowDecorations.h>
#include <opensef/OpenSEFAppKit.h>
#include <opensef/OpenSEFBackend.h>
#include <opensef/OpenSEFBase.h>

using namespace opensef;

int main() {
  std::cout << "╔════════════════════════════════════════════════════════╗"
            << std::endl;
  std::cout << "║       VitusOS Ares - openSEF Hello Window Test         ║"
            << std::endl;
  std::cout << "╚════════════════════════════════════════════════════════╝"
            << std::endl;
  std::cout << std::endl;

  // Connect to Wayland
  if (!OSFBackend::shared().connect()) {
    std::cerr << "Error: Could not connect to Wayland." << std::endl;
    std::cerr << "Make sure you're running in a Wayland session:" << std::endl;
    std::cerr << "  echo $XDG_SESSION_TYPE  # should say 'wayland'"
              << std::endl;
    return 1;
  }

  // Create window with XDG shell
  auto surface = OSFWaylandSurface::create(400, 300, "Hello VitusOS");
  if (!surface) {
    std::cerr << "Error: Failed to create window" << std::endl;
    OSFBackend::shared().disconnect();
    return 1;
  }

  // Fill with Space Orange (Ares primary color)
  OSFColor spaceOrange = OSFColors::primary();
  surface->draw(spaceOrange);

  std::cout << std::endl;
  std::cout << "╔════════════════════════════════════════════╗" << std::endl;
  std::cout << "║  Window created!                           ║" << std::endl;
  std::cout << "║                                            ║" << std::endl;
  std::cout << "║  Theme: Ares (The Martian)                 ║" << std::endl;
  std::cout << "║  Color: Space Orange #E85D04               ║" << std::endl;
  std::cout << "║                                            ║" << std::endl;
  std::cout << "║  Close the window to exit.                 ║" << std::endl;
  std::cout << "╚════════════════════════════════════════════╝" << std::endl;

  // Run event loop until window is closed
  OSFBackend::shared().run();

  // Cleanup
  surface.reset();
  OSFBackend::shared().disconnect();

  std::cout << "[openSEF] Goodbye!" << std::endl;
  return 0;
}

File: opensef/test/hello-window.m
/**
 * hello-window.m - openSEF Hello World test
 *
 * Creates a simple window to validate the framework builds and runs.
 */

#import <opensef/opensef.h>

int main(int argc, char *argv[]) {
  @autoreleasepool {
    NSLog(@"[openSEF] Hello Window Test");
    NSLog(@"[openSEF] Version: %s", OPENSEF_VERSION_STRING);

    // Connect to Wayland
    OSFBackend *backend = [OSFBackend sharedBackend];
    if (![backend connect]) {
      NSLog(@"[openSEF] ERROR: Failed to connect to Wayland");
      return 1;
    }

    // Create a simple window
    OSFWindow *window =
        [OSFWindow windowWithTitle:@"Hello VitusOS"
                             frame:CGRectMake(100, 100, 400, 300)];

    // Add glass background
    OSFGlassPanel *glass =
        [OSFGlassPanel glassPanelWithFrame:CGRectMake(0, 0, 400, 300)];
    glass.blurRadius = [OSFStyle blurRadiusMedium];
    [window setContentView:glass];

    // Add centered label
    OSFLabel *label = [[OSFLabel alloc] init];
    label.frame = CGRectMake(50, 120, 300, 60);
    label.text = @"Welcome to VitusOS";
    label.font = [OSFTypography headlineMedium];
    label.textColor = [OSFColors textPrimary];
    [glass addSubview:label];

    // Add subtitle
    OSFLabel *subtitle = [[OSFLabel alloc] init];
    subtitle.frame = CGRectMake(50, 170, 300, 30);
    subtitle.text = @"openSEF Framework is working!";
    subtitle.font = [OSFTypography bodyMedium];
    subtitle.textColor = [OSFColors textSecondary];
    [glass addSubview:subtitle];

    // Show window
    [window show];

    NSLog(@"[openSEF] Window created. Running event loop...");

    // Run event loop (blocks until window closed)
    [backend run];

    // Cleanup
    [backend disconnect];

    NSLog(@"[openSEF] Goodbye!");
    return 0;
  }
}
